daphne/src/roles/helper.rs

```diff
@@ -517,6 +518,16 @@ async fn finish_agg_job_and_aggregate<S>(
                     }));
                     inc_restart_metric.call_once(|| metrics.agg_job_put_span_retry_inc());
                 }
+                // This bucket belongs to a collected aggregate share.
+                (Err(MergeAggShareError::AlreadyCollected), reports) => {
+                    report_status.extend(reports.into_iter().map(|(report_id, _)| {
```

Comment:
Suggestion: We call `report_status.extend()` in every(?) arm of this match statement. What about having the match statement return a thing that we pass to `report_status.extend()`?

Reply:
it will be uglier, each of the things we are passing to extend is a different type so they won't unify. We could collect to a vec and return that or use dynamic dispatch, but both of those have performance costs and I don't think they improve readability that much